examples/feature-gated/Cargo.toml

```diff
@@ -0,0 +1,15 @@
+[package]
+name = "feature-gated"
+version = "0.2.0"
+authors = ["Samuel Pilz <samuel.pilz@posteo.net>"]
+edition = "2018"
+
+[dev-dependencies]
+mutagen = {path = "../../mutagen"}
```

Comment:
This could be an optional dependency.

Reply:
I tried that, but cargo responded with: "Dev-dependencies are not allowed to be optional: `mutagen`"

Reply:
https://github.com/rust-lang/cargo/issues/1596

Reply:
https://github.com/rust-lang/cargo/issues/1596

Reply:
One thing to consider: Mutagen would then be included when running `cargo build --release --all-features`. However, in the current form such call would fail. 

Please note that this is an example and people will figure out how they want to use mutagen themselves. If anyone really feels strongly about this issue, I can adapt the example.

Reply:
Nah, in that case it's OK.

Interesting. I think that's a missing cargo feature.